platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/synchronization/views/LearnerPushMerkleTreeView.java

```diff
@@ -273,4 +275,26 @@ public void releaseNode(final MerkleNode node) {
             node.release();
         }
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void recordHashStats(
+            @NonNull final ReconnectMapStats mapStats,
+            @NonNull final MerkleNode parent,
+            final int childIndex,
+            final boolean nodeAlreadyPresent) {
+        assert parent.isInternal();
```

Comment:
Do we use `assert` in product code? I do not think so.

Reply:
Yes, we do. At least we do in the former Platform Data code. And we do this a lot. This is done for performance reasons.

In this particular case, this is legit because the caller would only ever call this method for parents with actual children. Also, please see the javadoc for this method in the interface where it's defined.

Reply:
Just read the javadoc and it's not clear what in it mandates the use of assert. I don't think using assert is worth it here. I strongly recommend handling this as an exception instead of throwing a runtime error.

Reply:
I totally agree with that, and my preference has always been to use exceptions. But I've been told by @jasperpotts and @artemananiev that they use `assert` in Data code for the reasons I described above.

Since there's two reviewers who request using exceptions, I've updated the code accordingly.

PS. The javadoc for the method says `By the definition of this method, it is obvious that the parent is an internal node in the tree`. The `definition` is the first sentence of that javadoc which says `... about children of a given parent...`. And the method signature explicitly accepts a `childIndex` as the method operates on the children of the parent (again, per the definition of the method.)